app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/AppNavigation.kt

```diff
@@ -1,22 +1,120 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.lifecycle.viewmodel.compose.viewModel
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.rememberPagerState
+import com.vitorpamplona.amethyst.ui.dal.GlobalFeedFilter
+import com.vitorpamplona.amethyst.ui.dal.HomeConversationsFeedFilter
+import com.vitorpamplona.amethyst.ui.dal.HomeNewThreadFeedFilter
 import com.vitorpamplona.amethyst.ui.screen.AccountStateViewModel
+import com.vitorpamplona.amethyst.ui.screen.NostrGlobalFeedViewModel
+import com.vitorpamplona.amethyst.ui.screen.NostrHomeFeedViewModel
+import com.vitorpamplona.amethyst.ui.screen.NostrHomeRepliesFeedViewModel
 import com.vitorpamplona.amethyst.ui.screen.loggedIn.AccountViewModel
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.ChannelScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.ChatroomListScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.ChatroomScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.FiltersScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.HomeScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.NotificationScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.ProfileScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.SearchScreen
+import com.vitorpamplona.amethyst.ui.screen.loggedIn.ThreadScreen
 
+@OptIn(ExperimentalPagerApi::class)
 @Composable
 fun AppNavigation(
     navController: NavHostController,
     accountViewModel: AccountViewModel,
     accountStateViewModel: AccountStateViewModel,
     nextPage: String? = null
 ) {
+    val accountState by accountViewModel.accountLiveData.observeAsState()
+    val account = accountState?.account ?: return
+
+    GlobalFeedFilter.account = account
+    HomeNewThreadFeedFilter.account = account
+    HomeConversationsFeedFilter.account = account
+
+    val globalFeedViewModel: NostrGlobalFeedViewModel = viewModel()
+    val homeFeedViewModel: NostrHomeFeedViewModel = viewModel()
+    val homeRepliesFeedViewModel: NostrHomeRepliesFeedViewModel = viewModel()
+    val homePagerState = rememberPagerState()
+
     NavHost(navController, startDestination = Route.Home.route) {
-        Routes.forEach {
-            composable(it.route, it.arguments, content = it.buildScreen(accountViewModel, accountStateViewModel, navController))
+        Route.Search.let { route ->
+            composable(route.route, route.arguments, content = {
+                SearchScreen(
+                    accountViewModel = accountViewModel,
+                    feedViewModel = globalFeedViewModel,
+                    navController = navController,
+                    scrollToTop = it.arguments?.getBoolean("scrollToTop") ?: false
+                )
+            })
+        }
```

Comment:
This block replaces the `buildScreen` function. We need one per Route, but it's much simpler to build the screens directly.